src/service.js

```diff
@@ -37,77 +42,138 @@ export default class LambdaRestService {
+   * @param {boolean} [config.useScenarioName] - Whether to use scenario names for Cucumber tests
+   */
   constructor(options = {}, capabilities = {}, config = {}) {
     this._options = { ...DEFAULT_OPTIONS, ...options };
     this._capabilities = capabilities;
     this._config = config;
     this._testCnt = 0;
     this._failures = 0;
     // Cucumber specific
-    const strict = Boolean(this._config?.cucumberOpts && this._config?.cucumberOpts?.strict);
+    const strict = Boolean(
+      this._config?.cucumberOpts && this._config?.cucumberOpts?.strict
+    );
     // See https://github.com/cucumber/cucumber-js/blob/master/src/runtime/index.ts#L136
     if (strict) {
-      this._failureStatuses.push('pending');
+      this._failureStatuses.push("pending");
     }
   }
 
+  /**
+   * Hook called before test execution begins
+   * Initializes browser instance and resets scenario tracking
+   * @public
+   * @param {Object} caps - WebDriver capabilities
+   * @param {string[]} specs - Array of test spec file paths
+   * @param {Object} browser - WebDriver browser instance
+   */
   before(caps, specs, browser) {
     this._browser = browser;
     this._scenariosThatRan = [];
   }
 
+  /**
+   * Hook called before a new WebDriver session is created
+   * Configures service with session-specific settings and credentials
+   * @public
+   * @param {Object} config - WebdriverIO configuration object
+   * @param {Object} capabilities - WebDriver capabilities for the session
+   */
   beforeSession(config, capabilities) {
     this._config = { ...this._config, ...config };
     this._capabilities = { ...this._capabilities, ...capabilities };
     const lambdaCredentials = {
       username: this._config.user,
       accessKey: this._config.key,
-      isApp : false
+      isApp: false,
     };
 
-    if (this._config.product === 'appAutomation') {
+    if (this._config.product === "appAutomation") {
       lambdaCredentials.isApp = true;
     }
 
     if (this._config.logFile) {
       lambdaCredentials.logFile = this._config.logFile;
     }
-    if(this._config.ltErrorRemark ===true)
-    {
-      this._ltErrorRemark=true;
+    if (this._config.ltErrorRemark === true) {
+      this._ltErrorRemark = true;
     }
     // Cucumber specific option to set test name from scenario
-    if(this._config.useScenarioName === true)
-    {
-      this._useScenarioName=true;
+    if (this._config.useScenarioName === true) {
+      this._useScenarioName = true;
     }
 
-    this._isServiceEnabled = lambdaCredentials.username && lambdaCredentials.accessKey;
-    this._lambdaCredentials=lambdaCredentials;
-    
+    this._isServiceEnabled =
+      lambdaCredentials.username && lambdaCredentials.accessKey;
+    this._lambdaCredentials = lambdaCredentials;
   }
 
+  /**
+   * Hook called before each Cucumber scenario executes
+   * Sets the test title and session name based on scenario details
```

Comment:
We dont't need this 2nd line of JSdoc - `Sets the test title and session name based on scenario details`.
`beforeScenario` is a hook and the doc line sounds wrong, it will confuse the user.

Similar changes need to be done for other methods. Please verify for all methods manually before committing and lmk if you need help with any method.